contracts/token/ERC20/extensions/ERC4626.sol

```diff
@@ -33,18 +33,21 @@ abstract contract ERC4626 is ERC20, IERC4626 {
      * @dev Set the underlying asset contract. This must be an ERC20-compatible contract (ERC20 or ERC777).
      */
     constructor(IERC20 asset_) {
-        uint8 decimals_;
-        try IERC20Metadata(address(asset_)).decimals() returns (uint8 value) {
-            decimals_ = value;
-        } catch {
-            decimals_ = super.decimals();
-        }
-
+        (bool ok, bytes memory encodedDecimals) = address(asset_).call(
+            abi.encodeWithSelector(IERC20Metadata.decimals.selector)
+        );
+        _decimals = ok && encodedDecimals.length >= 32
+            ? uint8(abi.decode(encodedDecimals, (uint256)))
```

Comment:
Should we SafeCast ?

Reply:
We want to avoid reverts here. But I realize now that if the decimals is > 255 we should consider it an error and fallback to super.